cfme/containers/provider.py

```diff
@@ -267,3 +290,18 @@ def num_route(self):
     @num_route.variant('ui')
     def num_route_ui(self):
         return int(self.get_detail("Relationships", "Routes"))
+
+    @staticmethod
+    def configloader(prov_config, prov_key):
+        token_creds = OpenshiftProvider.process_credential_yaml_key(
+            prov_config['credentials'], cred_type='token')
+        return OpenshiftProvider(
+            name=prov_config['name'],
+            credentials={'token': token_creds},
+            key=prov_key,
+            zone=prov_config['server_zone'],
+            hostname=prov_config.get('hostname', None) or prov_config['ip_address'],
+            port=prov_config['port'],
+            provider_data=prov_config)
```

Comment:
This could be a class method. Then the OpenshiftProvider references won't be necessary. I guess it would then just need some "declarative" mapping of yaml keys -> constructor params and configloader would become generic.